actionpack/lib/action_dispatch/journey/nodes/node.rb

```diff
@@ -104,6 +104,15 @@ def group?; true; end
       end
 
       class Star < Unary # :nodoc:
+        attr_accessor :regexp
+
+        def initialize(left)
+          super(left)
+
+          # By default wildcard routes are non-greedy and must match something.
+          @regexp = /.+?/
```

Comment:
Should this regexp be pulled in to a constant?  There probably aren't *tons* of `Star` nodes in the world, but we could reduce the number of objects allocated by moving this to a constant.

Reply:
I believe this will use a single object anyways since it's a regex literal

```
$ ruby -e '5.times { p /foo/.object_id }'
60
60
60
60
60
```

Reply:
🤦🏻‍♂️ you're totally right!